application/backend/src/main/resources/apps/typerefinery/components/content/ticker/script.js

```diff
@@ -48,43 +36,166 @@ function updateTickerComponent(tickerData, id) {
             <div class="icon ${tickerData.icon}"></div>
     `;
 
-    const component = document.getElementById(id);
     component.innerHTML = tickerHtmlWithJsonValue;
-
+  // }
 }
 
-function tickerComponentMounted(id, component) {
 
-    var defaultData = {
-        "title": component.getElementsByClassName("heading")[0].innerHTML,
-        "value": component.getElementsByClassName("value")[0].innerHTML,
-        "icon": component.getElementsByClassName("icon")[0].innerHTML,
-        "indicatorType": component.getElementsByClassName("indicator-type")[0].innerHTML,
-        "indicatorValue": component.getElementsByClassName("indicator-value")[0].innerHTML
+//Update ticker using tms connection and its payload
+function tickerComponentConnectedViaTMS(topic, host, component) {  
+  setTimeout(() => {
+    connectTMS( topic,host,component);
+  }, 5000);
+}
+// TMS connection
+function connectTMS(topic, host,component) {
+  var topic = component.getAttribute("data-topic")
+  window.addEventListener(
+    window.MessageService.Client.events.ERROR,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.ERROR}`,
+        message.detail
+      );
     }
-
-    // getting the dataSource of the component
-    var dataSourcePath = component.getAttribute("data-source") || "";
-
-    // getting the dataSource of the component
-    var dataSourceReferenceTime = component.getAttribute("data-source-refresh-time") || 5;
-    // cast to number (ms)
-    if (dataSourceReferenceTime) {
-        dataSourceReferenceTime = Number(dataSourceReferenceTime) * 1000;
-    } else {
-        dataSourceReferenceTime = 5000;
+  );
+
+  // listen when the client is ready
+  window.addEventListener(
+    window.MessageService.Client.events.READY,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.READY}`,
+        message.detail
+      );
     }
-
-    // Rendering the template
-    getDataFromDataSource(defaultData, dataSourcePath, id, dataSourceReferenceTime);
-
+  );
+
+  // listen for client id update
+  window.addEventListener(
+    window.MessageService.Client.events.CLIENT_ID,
+    function (message) {
+      console.log(
+        `ui event: ${window.MessageService.Client.events.CLIENT_ID}`,
+        message.detail
+      );
+      document.querySelector("#ws-id").textContent = message.detail;
+    }
+  );
+
+  // listen for messages
+  window.addEventListener(
```

Comment:
In this event listeners I can see many un used code, please do remove them

Reply:
Removed